### PR TITLE
Update DPG ResponseClassifiers

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/autorest.csharp/issues/2006

### CODE REVIEW

File: test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs [hunk 1]
```diff
@@ -159,47 +159,42 @@ public virtual Response<bool> Head404(RequestContext context = null)
...
         }
 
-        private sealed class ResponseClassifier200To300400To500 : ResponseClassifier
+        private sealed class ResponseClassifier200To300400To500Override : ResponseClassifier
         {
-            private static ResponseClassifier _instance;
-            public static ResponseClassifier Instance => _instance ??= new ResponseClassifier200To300400To500();
             public override bool IsErrorResponse(HttpMessage message)
             {
                 return message.Response.Status switch
```
Comment (ShivangiReja) [inline test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs:200]: [Here](https://github.com/Azure/autorest.csharp/blob/ebc6eede5294c2b0300bf86b28e852c5fa3a4f03/test/TestProjects/HeadAsBooleanTrue-LowLevel/Generated/HttpSuccessClient.cs#L196-L210) is the generated code for the `ResponseClassifier200To300400To500` bases on https://github.com/Azure/autorest.csharp/issues/2006#issuecomment-1055887869

File: test/TestProjects/Accessibility-LowLevel/Generated/AccessibilityClient.cs [hunk 2]
```diff
@@ -147,22 +146,10 @@ internal HttpMessage CreateOperationInternalRequest(RequestContent content, Requ
...
-            {
-                return message.Response.Status switch
-                {
-                    200 => false,
-                    _ => true
-                };
-            }
-        }
+        private static ResponseClassifier _responseClassifier200;
+        private static ResponseClassifier ResponseClassifier200 => _responseClassifier200 ??= new CoreResponseClassifier(stackalloc int[] { 200 });
```
Comment (ShivangiReja) [inline test/TestProjects/Accessibility-LowLevel/Generated/AccessibilityClient.cs:153]: [Here](https://github.com/Azure/autorest.csharp/blob/ebc6eede5294c2b0300bf86b28e852c5fa3a4f03/test/TestProjects/Accessibility-LowLevel/Generated/AccessibilityClient.cs#L152-L153) is the generated code for `ResponseClassifier200` bases on https://github.com/Azure/autorest.csharp/issues/2006#issuecomment-1055883992

File: test/TestServerProjectsLowLevel/head/Generated/HttpSuccessClient.cs [hunk 3]
```diff
@@ -159,70 +159,43 @@ public virtual Response Head404(RequestContext context = null)
...
-                    204 => false,
-                    404 => false,
-                    _ => true
-                };
-            }
-        }
+        private static ResponseClassifier _responseClassifier200404;
+        private static ResponseClassifier ResponseClassifier200404 => _responseClassifier200404 ??= new CoreResponseClassifier(stackalloc int[] { 200, 404 });
+        private static ResponseClassifier _responseClassifier204404;
+        private static ResponseClassifier ResponseClassifier204404 => _responseClassifier204404 ??= new CoreResponseClassifier(stackalloc int[] { 204, 404 });
```
Comment (ShivangiReja) [inline test/TestServerProjectsLowLevel/head/Generated/HttpSuccessClient.cs:199]: @annelo-msft Test case for [Head404Async()](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServerLowLevel.Tests/head.cs#L30) method is failing with these updates. Check [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1404658&view=logs&j=b5cdfb1e-c180-561f-918e-a169e451719e&t=d557719e-b1de-5962-364a-0820efa4a6bb&l=47) in pipeline for the error message.

Comment (ShivangiReja) [reply test/TestServerProjectsLowLevel/head/Generated/HttpSuccessClient.cs]: This is issue fixed [here](https://github.com/Azure/azure-sdk-for-net/pull/27334).

**Comment by annelo-msft** [comment]
Once https://github.com/Azure/azure-sdk-for-net/pull/27379 merges, we will need to change the CoreResponseClassifier name to StatusCodeClassifier.  Probably better to hold this until that happens.

**Review by annelo-msft — APPROVED** [review]

**Pull request merged**